sermant-plugins/sermant-flowcontrol/flowcontrol-common/src/main/java/io/sermant/flowcontrol/common/xds/retry/ExceptionRetryConditionType.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.flowcontrol.common.xds.retry;
+
+import io.sermant.core.utils.StringUtils;
+import io.sermant.flowcontrol.common.xds.retry.condition.ConnectErrorRetryCondition;
+import io.sermant.flowcontrol.common.xds.retry.condition.ResetBeforeRequestErrorCondition;
+import io.sermant.flowcontrol.common.xds.retry.condition.ResetErrorCondition;
+import io.sermant.flowcontrol.common.xds.retry.condition.ServerExceptionCondition;
+import io.sermant.flowcontrol.common.xds.retry.condition.SpecificHeaderNameErrorRetryCondition;
+
+import java.util.Optional;
+
+/**
+ * Retry Condition Manager
+ *
+ * @author zhp
+ * @since 2024-11-29
+ */
+public enum ExceptionRetryConditionType {
+    /**
+     * The type of conditional judgment for server errors
+     */
+    SERVER_ERROR("5xx", new ServerExceptionCondition()),
+
+    /**
+     * The type of conditional judgment for reset errors
+     */
+    RESET_ERROR("reset", new ResetErrorCondition()),
+
+    /**
+     * The type of conditional judgment for resetting errors before request
+     */
+    RESET_BEFORE_REQUEST_ERROR("reset-before-request", new ResetBeforeRequestErrorCondition()),
+
+    /**
+     * The type of conditional judgment for connect errors
+     */
+    CONNECT_ERROR("connect-failure", new ConnectErrorRetryCondition()),
+
+    /**
+     * The type of conditional judgment for Specify response headers
+     */
+    SPECIFIC_HEADER_NAME_ERROR("retriable-headers", new SpecificHeaderNameErrorRetryCondition());
```

Comment:
it is better to make the type name and class name same as condition name

Reply:
done